routing/missioncontrol_store.go

```diff
@@ -339,12 +339,52 @@ func (b *missionControlStore) storeResults() error {
 	}
 
 	var (
-		keys       *list.List
-		keysMap    map[string]struct{}
+		newKeys    map[string]*paymentResult
+		delKeys    []string
 		storeCount int
 		pruneCount int
 	)
 
+	// Create a deduped list of new entries.
+	newKeys = make(map[string]*paymentResult, l.Len())
+	for e := l.Front(); e != nil; {
```

Comment:
use `for e := l.Front(); e != nil; e = e.Next() {` then dont have to call e.Next manually?